pkg/client/client_venconn.go

```diff
@@ -0,0 +1,215 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/jetstack/preflight/api"
+	"github.com/jetstack/preflight/pkg/version"
+	venapi "github.com/jetstack/venafi-connection-lib/api/v1alpha1"
+	"github.com/jetstack/venafi-connection-lib/venafi_client"
+	"github.com/jetstack/venafi-connection-lib/venafi_client/auth"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	ctrlruntimelog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type VenConnClient struct {
+	agentMetadata *api.AgentMetadata
+	connHandler   venafi_client.ConnectionHandler
+	installNS     string       // Namespace in which the agent is running in.
+	venConnName   string       // Name of the VenafiConnection resource to use.
+	venConnNS     string       // Namespace of the VenafiConnection resource to use.
+	client        *http.Client // Used to make HTTP requests to Venafi Cloud.
+}
+
+// NewVenConnClient lets you make requests to the Venafi Cloud backend using the
+// given VenafiConnection resource.
+//
+// You need to call Start to start watching the VenafiConnection resource. If
+// you don't, the client will be unable to find the VenafiConnection that you
+// are referring to as its client-go cache will remain empty.
+//
+// The http.Client is used for Venafi and Vault, not for Kubernetes. The
+// `installNS` is the namespace in which the agent is running in. The passed
+// `restcfg` is not mutated. `trustedCAs` is only used for connecting to Venafi
+// Cloud and Vault and can be left nil.
+func NewVenConnClient(restcfg *rest.Config, agentMetadata *api.AgentMetadata, installNS, venConnName, venConnNS string, trustedCAs *x509.CertPool) (*VenConnClient, error) {
+	// TODO(mael): The rest of the codebase uses the standard "log" package,
+	// venafi-connection-lib uses "go-logr/logr", and client-go uses "klog". We
+	// should standardize on one of them, probably "slog".
+	ctrlruntimelog.SetLogger(logr.Logger{})
```

Comment:
This TODO has been addressed in https://github.com/jetstack/jetstack-secure/pull/626: 
 * https://github.com/jetstack/jetstack-secure/pull/626